projects/igniteui-angular/src/lib/grids/grid-navigation.service.ts

```diff
@@ -553,6 +555,15 @@ export class IgxGridNavigationService {
             });
         this.horizontalScroll(rowIndex).scrollTo(unpinnedIndex);
     }
+
+    private focusGridWithoutScrolling() {
```

Comment:
The base grid navigation service should not contains checks for hierarchicalgrid specific properties.
HierarchicalgGrid specific code should be in hierarchical-grid-navigation.service.
Make this function protected and overwrite it in the hierarchical grid navigation service with its specific code.